Sources/SourceControl/RepositoryManager.swift

```diff
@@ -342,25 +301,34 @@ public class RepositoryManager {
         to destinationPath: AbsolutePath,
         editable: Bool
     ) throws {
-        try provider.cloneCheckout(
-            repository: handle.repository,
-            at: path.appending(handle.subpath),
-            to: destinationPath,
-            editable: editable)
+        let lock = FileLock(name: handle.repository.basename, cachePath: self.path)
+        // FIXME: Workaround for `FileLock` only working on `LocaFileSystem`
+        if localFileSystem.exists(self.path) {
+            try lock.withLock {
+                try lock.withLock {
```

Comment:
Not sure I understand why we are doing this nesting

Reply:
When I added the workaround to only lock when we are on the local filesystem, I probably copy pasted the block including the lock and added another lock.